clientv3/integration: increase txn timeout limit

gyuho opened the pull request:
Fix https://github.com/coreos/etcd/issues/5416.

gyuho commented:
```
[DEBUG] 2016-06-02 00:04:59.826239284 +0000 UTC kv.Txn starting!
2016-06-02 00:04:59.848624 E | rafthttp: failed to write 94e5d7e43cbbf7c7 on stream Message (write tcp 127.0.0.1:21111->127.0.0.1:35833: write: broken pipe)
2016-06-02 00:04:59.848679 I | rafthttp: peer 94e5d7e43cbbf7c7 became inactive
2016-06-02 00:04:59.849499 W | rafthttp: lost the TCP streaming connection with peer 94e5d7e43cbbf7c7 (stream Message writer)
2016-06-02 00:04:59.913138 E | rafthttp: failed to dial 94e5d7e43cbbf7c7 on stream MsgApp v2 (dial tcp 127.0.0.1:21109: getsockopt: connection refused)
2016-06-02 00:04:59.913198 I | rafthttp: peer 94e5d7e43cbbf7c7 became inactive
2016-06-02 00:05:01.318176 W | rafthttp: lost the TCP streaming connection with peer 94e5d7e43cbbf7c7 (stream MsgApp v2 writer)
2016-06-02 00:05:01.331652 W | etcdserver: failed to send out heartbeat on time (exceeded the 10ms timeout for 3.050851ms)
2016-06-02 00:05:01.331745 W | etcdserver: server is likely overloaded
2016-06-02 00:05:01.331794 W | etcdserver: failed to send out heartbeat on time (exceeded the 10ms timeout for 3.208878ms)
2016-06-02 00:05:01.331825 W | etcdserver: server is likely overloaded
2016-06-02 00:05:02.254015 W | etcdserver: failed to send out heartbeat on time (exceeded the 10ms timeout for 5.275378ms)
2016-06-02 00:05:02.254089 W | etcdserver: server is likely overloaded
2016-06-02 00:05:02.257564 W | etcdserver: failed to send out heartbeat on time (exceeded the 10ms timeout for 8.836674ms)
2016-06-02 00:05:02.261354 W | etcdserver: server is likely overloaded
2016-06-02 00:05:03.807011 W | etcdserver: failed to reach the peerURL(http://127.0.0.1:21109) of member 94e5d7e43cbbf7c7 (Get http://127.0.0.1:21109/version: dial tcp 127.0.0.1:21109: getsockopt: connection refused)
2016-06-02 00:05:03.810441 W | etcdserver: cannot get the version of member 94e5d7e43cbbf7c7 (Get http://127.0.0.1:21109/version: dial tcp 127.0.0.1:21109: getsockopt: connection refused)
2016-06-02 00:05:06.276969 W | rafthttp: lost the TCP streaming connection with peer 94e5d7e43cbbf7c7 (stream MsgApp v2 writer)
2016-06-02 00:05:06.277258 W | rafthttp: lost the TCP streaming connection with peer 94e5d7e43cbbf7c7 (stream Message writer)
[DEBUG] 2016-06-02 00:05:07.379072576 +0000 UTC connWait c.lastConnErr grpc: timed out trying to connect
2016-06-02 00:05:07.811783 W | etcdserver: failed to reach the peerURL(http://127.0.0.1:21109) of member 94e5d7e43cbbf7c7 (Get http://127.0.0.1:21109/version: dial tcp 127.0.0.1:21109: getsockopt: connection refused)
2016-06-02 00:05:07.811837 W | etcdserver: cannot get the version of member 94e5d7e43cbbf7c7 (Get http://127.0.0.1:21109/version: dial tcp 127.0.0.1:21109: getsockopt: connection refused)
[DEBUG] 2016-06-02 00:05:10.826283022 +0000 UTC timed out waiting for txn to fail
2016-06-02 00:05:11.813448 W | etcdserver: failed to reach the peerURL(http://127.0.0.1:21109) of member 94e5d7e43cbbf7c7 (Get http://127.0.0.1:21109/version: dial tcp 127.0.0.1:21109: getsockopt: connection refused)
2016-06-02 00:05:11.813525 W | etcdserver: cannot get the version of member 94e5d7e43cbbf7c7 (Get http://127.0.0.1:21109/version: dial tcp 127.0.0.1:21109: getsockopt: connection refused)
[DEBUG] 2016-06-02 00:05:11.90833063 +0000 UTC connWait c.lastConnErr grpc: timed out trying to connect
[DEBUG] 2016-06-02 00:05:11.908408404 +0000 UTC kv.Txn error: rpc error: code = 2 desc = grpc: the client connection is closing
2016-06-02 00:05:11.909870 I | integration: terminating node7564707074000347284 (unix://localhost:node7564707074000347284.sock)
2016-06-02 00:05:11.916167 I | integration: terminated node7564707074000347284 (unix://localhost:node7564707074000347284.sock)
```

gyuho commented:
/cc @xiang90 @heyitsanthony This happens even with election timeout 600.

As you can see it in the logs, `kv.Txn.Commit` actually returns the expected error. And we have 12 seconds to receive this error, otherwise times out. And it took 13 seconds to return that error, so it timed out.

Simply increasing the timeout for Semaphore CI should fix the issue.

gyuho commented:
Reproducible at:
- https://semaphoreci.com/coreos/etcd/branches/pull-request-5523/
- https://github.com/coreos/etcd/pull/5523

xiang90 commented:
@gyuho We need to know why exactly it needs 12 seconds to get the expected response I think.

xiang90 commented on clientv3/integration/txn_test.go:
```diff
@@ -74,7 +74,7 @@ func TestTxnWriteFail(t *testing.T) {
 	dialTimeout := 5 * time.Second
 	select {
-	case <-time.After(2*dialTimeout + time.Second):
+	case <-time.After(3*dialTimeout + time.Second):
```
@heyitsanthony We actually only need 1 dailtimeout in theory, right?

gyuho replied:
Actually, we don't have control in the dial in this case I think. (we use `context.TODO`). It looks like gRPC sets its own timeout when we don't pass timeout https://github.com/grpc/grpc-go/blob/master/clientconn.go#L452-L459.

xiang90 replied:
No. We control it. The timeout should be passed into gRPC, which is 5 seconds by default.

gyuho replied:
You are right. We are passing 5 seconds, in integration package.

heyitsanthony replied:
@xiang90 one dialtimeout + time it takes to realize it's disconnected

gyuho commented:
@xiang90 

First we don't set timeout for context, so it's up to our client reconnection logic to return error. Even without test failure, it takes more than 7 seconds:

```
[DEBUG] 2016-06-02 00:03:08.394854419 +0000 UTC kv.Txn stop done!
[DEBUG] 2016-06-02 00:03:08.395013852 +0000 UTC kv.Txn starting!
2016-06-02 00:03:08.408078 E | rafthttp: failed to write 577295b580b977df on stream Message (write tcp 127.0.0.1:21003->127.0.0.1:49925: write: broken pipe)
2016-06-02 00:03:08.408173 I | rafthttp: peer 577295b580b977df became inactive
2016-06-02 00:03:08.409862 W | rafthttp: lost the TCP streaming connection with peer 577295b580b977df (stream Message writer)
2016-06-02 00:03:08.478239 E | rafthttp: failed to dial 577295b580b977df on stream MsgApp v2 (dial tcp 127.0.0.1:21001: getsockopt: connection refused)
2016-06-02 00:03:08.478413 I | rafthttp: peer 577295b580b977df became inactive
2016-06-02 00:03:08.786150 I | api: enabled capabilities for version 3.0.0
2016-06-02 00:03:09.669549 W | rafthttp: lost the TCP streaming connection with peer 577295b580b977df (stream MsgApp v2 writer)
2016-06-02 00:03:12.351958 W | etcdserver: failed to reach the peerURL(http://127.0.0.1:21001) of member 577295b580b977df (Get http://127.0.0.1:21001/version: dial tcp 127.0.0.1:21001: getsockopt: connection refused)
2016-06-02 00:03:12.352229 W | etcdserver: cannot get the version of member 577295b580b977df (Get http://127.0.0.1:21001/version: dial tcp 127.0.0.1:21001: getsockopt: connection refused)
2016-06-02 00:03:14.859366 W | rafthttp: lost the TCP streaming connection with peer 577295b580b977df (stream MsgApp v2 writer)
2016-06-02 00:03:14.860172 W | rafthttp: lost the TCP streaming connection with peer 577295b580b977df (stream Message writer)
[DEBUG] 2016-06-02 00:03:15.328753074 +0000 UTC kv.Txn error: rpc error: code = 2 desc = grpc: the client connection is closing
[DEBUG] 2016-06-02 00:03:15.328822267 +0000 UTC kv.Txn sent to donec
[DEBUG] 2016-06-02 00:03:15.328861564 +0000 UTC donec
[DEBUG] 2016-06-02 00:03:15.328863452 +0000 UTC connWait c.lastConnErr grpc: timed out trying to connect
```

And regarding 12 seconds, it depends on how gRPC handles connection error. Code is here https://github.com/grpc/grpc-go/blob/master/clientconn.go#L431. I think it keeps retrying with sleep time, taking it several seconds.

xiang90 commented:
@gyuho Can you post your patched version of test somewhere? It is not clear to me where the debugging ling got printed.

xiang90 commented:
Also can you post the full log from the testing start to its end?

gyuho commented:
Full logs here:

```
=== RUN   TestTxnWriteFail
2016-06-02 00:04:59.547337 I | integration: launching node6962765416249702947 (unix://localhost:node6962765416249702947.sock)
2016-06-02 00:04:59.548744 I | integration: launching node7564707074000347284 (unix://localhost:node7564707074000347284.sock)
2016-06-02 00:04:59.549868 I | integration: launching node1077664708996568847 (unix://localhost:node1077664708996568847.sock)
2016-06-02 00:04:59.568521 I | etcdserver: name = node6962765416249702947
2016-06-02 00:04:59.568596 I | etcdserver: data dir = /tmp/etcd678854046
2016-06-02 00:04:59.568649 I | etcdserver: member dir = /tmp/etcd678854046/member
2016-06-02 00:04:59.568692 I | etcdserver: heartbeat = 10ms
2016-06-02 00:04:59.568729 I | etcdserver: election = 100ms
2016-06-02 00:04:59.568767 I | etcdserver: snapshot count = 0
2016-06-02 00:04:59.568817 I | etcdserver: advertise client URLs = http://127.0.0.1:21114
2016-06-02 00:04:59.568864 I | etcdserver: initial advertise peer URLs = http://127.0.0.1:21113
2016-06-02 00:04:59.568947 I | etcdserver: initial cluster = node1077664708996568847=http://127.0.0.1:21111,node6962765416249702947=http://127.0.0.1:21113,node7564707074000347284=http://127.0.0.1:21109
2016-06-02 00:04:59.569984 I | etcdserver: name = node7564707074000347284
2016-06-02 00:04:59.570031 I | etcdserver: data dir = /tmp/etcd410208396
2016-06-02 00:04:59.570061 I | etcdserver: member dir = /tmp/etcd410208396/member
2016-06-02 00:04:59.570086 I | etcdserver: heartbeat = 10ms
2016-06-02 00:04:59.570105 I | etcdserver: election = 100ms
2016-06-02 00:04:59.570124 I | etcdserver: snapshot count = 0
2016-06-02 00:04:59.570153 I | etcdserver: advertise client URLs = http://127.0.0.1:21110
2016-06-02 00:04:59.570195 I | etcdserver: initial advertise peer URLs = http://127.0.0.1:21109
2016-06-02 00:04:59.570245 I | etcdserver: initial cluster = node1077664708996568847=http://127.0.0.1:21111,node6962765416249702947=http://127.0.0.1:21113,node7564707074000347284=http://127.0.0.1:21109
2016-06-02 00:04:59.570883 I | etcdserver: name = node1077664708996568847
2016-06-02 00:04:59.570931 I | etcdserver: data dir = /tmp/etcd502473851
2016-06-02 00:04:59.570972 I | etcdserver: member dir = /tmp/etcd502473851/member
2016-06-02 00:04:59.571004 I | etcdserver: heartbeat = 10ms
2016-06-02 00:04:59.571035 I | etcdserver: election = 100ms
2016-06-02 00:04:59.571066 I | etcdserver: snapshot count = 0
2016-06-02 00:04:59.571105 I | etcdserver: advertise client URLs = http://127.0.0.1:21112
2016-06-02 00:04:59.571142 I | etcdserver: initial advertise peer URLs = http://127.0.0.1:21111
2016-06-02 00:04:59.571212 I | etcdserver: initial cluster = node1077664708996568847=http://127.0.0.1:21111,node6962765416249702947=http://127.0.0.1:21113,node7564707074000347284=http://127.0.0.1:21109
2016-06-02 00:04:59.590263 I | etcdserver: starting member 94e5d7e43cbbf7c7 in cluster 2b55eb8bf5655e7d
2016-06-02 00:04:59.590452 I | raft: 94e5d7e43cbbf7c7 became follower at term 0
2016-06-02 00:04:59.590502 I | raft: newRaft 94e5d7e43cbbf7c7 [peers: [], term: 0, commit: 0, applied: 0, lastindex: 0, lastterm: 0]
2016-06-02 00:04:59.590536 I | raft: 94e5d7e43cbbf7c7 became follower at term 1
2016-06-02 00:04:59.592341 I | rafthttp: starting peer 8996f73a60a6d61f...
2016-06-02 00:04:59.592643 I | rafthttp: started HTTP pipelining with peer 8996f73a60a6d61f
2016-06-02 00:04:59.592678 I | etcdserver: starting member 8996f73a60a6d61f in cluster 2b55eb8bf5655e7d
2016-06-02 00:04:59.592882 I | raft: 8996f73a60a6d61f became follower at term 0
2016-06-02 00:04:59.592954 I | raft: newRaft 8996f73a60a6d61f [peers: [], term: 0, commit: 0, applied: 0, lastindex: 0, lastterm: 0]
2016-06-02 00:04:59.593004 I | raft: 8996f73a60a6d61f became follower at term 1
2016-06-02 00:04:59.596208 I | rafthttp: starting peer 94e5d7e43cbbf7c7...
2016-06-02 00:04:59.596539 I | rafthttp: started peer 8996f73a60a6d61f
2016-06-02 00:04:59.596808 I | rafthttp: added peer 8996f73a60a6d61f
2016-06-02 00:04:59.596858 I | rafthttp: started HTTP pipelining with peer 94e5d7e43cbbf7c7
2016-06-02 00:04:59.597054 I | rafthttp: started streaming with peer 8996f73a60a6d61f (writer)
2016-06-02 00:04:59.597147 I | rafthttp: started streaming with peer 8996f73a60a6d61f (writer)
2016-06-02 00:04:59.597300 I | rafthttp: started streaming with peer 8996f73a60a6d61f (stream Message reader)
2016-06-02 00:04:59.597735 I | rafthttp: started streaming with peer 8996f73a60a6d61f (stream MsgApp v2 reader)
2016-06-02 00:04:59.598480 I | rafthttp: starting peer bf1facf945c123f0...
2016-06-02 00:04:59.599104 I | rafthttp: started HTTP pipelining with peer bf1facf945c123f0
2016-06-02 00:04:59.602128 I | rafthttp: started peer 94e5d7e43cbbf7c7
2016-06-02 00:04:59.602698 I | rafthttp: added peer 94e5d7e43cbbf7c7
2016-06-02 00:04:59.603418 I | rafthttp: starting peer bf1facf945c123f0...
2016-06-02 00:04:59.604189 I | rafthttp: started HTTP pipelining with peer bf1facf945c123f0
2016-06-02 00:04:59.605604 I | rafthttp: started peer bf1facf945c123f0
2016-06-02 00:04:59.605959 I | rafthttp: added peer bf1facf945c123f0
2016-06-02 00:04:59.606039 I | etcdserver: set snapshot count to default 10000
2016-06-02 00:04:59.606106 I | etcdserver: starting server... [version: 3.0.0-beta.0+git, cluster version: to_be_decided]
2016-06-02 00:04:59.608434 I | rafthttp: started streaming with peer bf1facf945c123f0 (writer)
2016-06-02 00:04:59.608631 I | rafthttp: started streaming with peer bf1facf945c123f0 (writer)
2016-06-02 00:04:59.608808 I | rafthttp: started streaming with peer bf1facf945c123f0 (stream MsgApp v2 reader)
2016-06-02 00:04:59.609186 I | rafthttp: started streaming with peer 94e5d7e43cbbf7c7 (writer)
2016-06-02 00:04:59.609247 I | rafthttp: started streaming with peer bf1facf945c123f0 (stream Message reader)
2016-06-02 00:04:59.609346 I | rafthttp: started streaming with peer 94e5d7e43cbbf7c7 (stream MsgApp v2 reader)
2016-06-02 00:04:59.609555 I | rafthttp: started streaming with peer 94e5d7e43cbbf7c7 (stream Message reader)
2016-06-02 00:04:59.609921 I | rafthttp: started streaming with peer bf1facf945c123f0 (writer)
2016-06-02 00:04:59.609988 I | rafthttp: started streaming with peer bf1facf945c123f0 (writer)
2016-06-02 00:04:59.610075 I | rafthttp: started streaming with peer bf1facf945c123f0 (stream Message reader)
2016-06-02 00:04:59.611083 I | integration: launched node7564707074000347284 (unix://localhost:node7564707074000347284.sock)
2016-06-02 00:04:59.611164 I | rafthttp: started streaming with peer 94e5d7e43cbbf7c7 (writer)
2016-06-02 00:04:59.612686 I | membership: added member 8996f73a60a6d61f [http://127.0.0.1:21113] to cluster 2b55eb8bf5655e7d
2016-06-02 00:04:59.613799 I | rafthttp: started peer bf1facf945c123f0
2016-06-02 00:04:59.614034 I | rafthttp: added peer bf1facf945c123f0
2016-06-02 00:04:59.614096 I | etcdserver: set snapshot count to default 10000
2016-06-02 00:04:59.614146 I | etcdserver: starting server... [version: 3.0.0-beta.0+git, cluster version: to_be_decided]
2016-06-02 00:04:59.615183 I | etcdserver: starting member bf1facf945c123f0 in cluster 2b55eb8bf5655e7d
2016-06-02 00:04:59.615477 I | raft: bf1facf945c123f0 became follower at term 0
2016-06-02 00:04:59.615555 I | raft: newRaft bf1facf945c123f0 [peers: [], term: 0, commit: 0, applied: 0, lastindex: 0, lastterm: 0]
2016-06-02 00:04:59.615604 I | raft: bf1facf945c123f0 became follower at term 1
2016-06-02 00:04:59.616175 I | membership: added member 94e5d7e43cbbf7c7 [http://127.0.0.1:21109] to cluster 2b55eb8bf5655e7d
2016-06-02 00:04:59.617725 I | raft: 8996f73a60a6d61f is starting a new election at term 1
2016-06-02 00:04:59.617830 I | raft: 8996f73a60a6d61f became candidate at term 2
2016-06-02 00:04:59.617880 I | raft: 8996f73a60a6d61f received vote from 8996f73a60a6d61f at term 2
2016-06-02 00:04:59.618049 I | raft: 8996f73a60a6d61f [logterm: 1, index: 3] sent vote request to bf1facf945c123f0 at term 2
2016-06-02 00:04:59.618109 I | raft: 8996f73a60a6d61f [logterm: 1, index: 3] sent vote request to 94e5d7e43cbbf7c7 at term 2
2016-06-02 00:04:59.620203 I | rafthttp: peer 94e5d7e43cbbf7c7 became active
2016-06-02 00:04:59.620293 I | rafthttp: established a TCP streaming connection with peer 94e5d7e43cbbf7c7 (stream Message writer)
2016-06-02 00:04:59.620380 I | rafthttp: starting peer 8996f73a60a6d61f...
2016-06-02 00:04:59.620780 I | rafthttp: started HTTP pipelining with peer 8996f73a60a6d61f
2016-06-02 00:04:59.621694 I | rafthttp: established a TCP streaming connection with peer 94e5d7e43cbbf7c7 (stream MsgApp v2 writer)
2016-06-02 00:04:59.622141 I | membership: added member bf1facf945c123f0 [http://127.0.0.1:21111] to cluster 2b55eb8bf5655e7d
2016-06-02 00:04:59.623364 I | membership: added member 8996f73a60a6d61f [http://127.0.0.1:21113] to cluster 2b55eb8bf5655e7d
2016-06-02 00:04:59.624258 I | rafthttp: started streaming with peer 8996f73a60a6d61f (writer)
2016-06-02 00:04:59.624552 I | rafthttp: started streaming with peer bf1facf945c123f0 (stream MsgApp v2 reader)
2016-06-02 00:04:59.639368 I | raft: 94e5d7e43cbbf7c7 is starting a new election at term 1
2016-06-02 00:04:59.639815 I | raft: 94e5d7e43cbbf7c7 became candidate at term 2
2016-06-02 00:04:59.639932 I | raft: 94e5d7e43cbbf7c7 received vote from 94e5d7e43cbbf7c7 at term 2
2016-06-02 00:04:59.640040 I | rafthttp: started peer 8996f73a60a6d61f
2016-06-02 00:04:59.640708 I | rafthttp: peer 8996f73a60a6d61f became active
2016-06-02 00:04:59.640883 I | rafthttp: established a TCP streaming connection with peer 8996f73a60a6d61f (stream MsgApp v2 reader)
2016-06-02 00:04:59.641555 I | rafthttp: added peer 8996f73a60a6d61f
2016-06-02 00:04:59.642163 I | rafthttp: starting peer 94e5d7e43cbbf7c7...
2016-06-02 00:04:59.642320 I | rafthttp: established a TCP streaming connection with peer 8996f73a60a6d61f (stream Message reader)
2016-06-02 00:04:59.643394 I | membership: added member 94e5d7e43cbbf7c7 [http://127.0.0.1:21109] to cluster 2b55eb8bf5655e7d
2016-06-02 00:04:59.643539 I | rafthttp: started HTTP pipelining with peer 94e5d7e43cbbf7c7
2016-06-02 00:04:59.644027 I | rafthttp: started streaming with peer 8996f73a60a6d61f (writer)
2016-06-02 00:04:59.646279 I | rafthttp: started streaming with peer 8996f73a60a6d61f (stream MsgApp v2 reader)
2016-06-02 00:04:59.647132 I | raft: 94e5d7e43cbbf7c7 [logterm: 1, index: 3] sent vote request to bf1facf945c123f0 at term 2
2016-06-02 00:04:59.647385 I | raft: 94e5d7e43cbbf7c7 [logterm: 1, index: 3] sent vote request to 8996f73a60a6d61f at term 2
2016-06-02 00:04:59.647681 I | rafthttp: started streaming with peer 8996f73a60a6d61f (stream Message reader)
2016-06-02 00:04:59.648349 I | membership: added member bf1facf945c123f0 [http://127.0.0.1:21111] to cluster 2b55eb8bf5655e7d
2016-06-02 00:04:59.648922 I | rafthttp: started peer 94e5d7e43cbbf7c7
2016-06-02 00:04:59.649197 I | rafthttp: added peer 94e5d7e43cbbf7c7
2016-06-02 00:04:59.649254 I | etcdserver: set snapshot count to default 10000
2016-06-02 00:04:59.649300 I | etcdserver: starting server... [version: 3.0.0-beta.0+git, cluster version: to_be_decided]
2016-06-02 00:04:59.650948 I | integration: launched node6962765416249702947 (unix://localhost:node6962765416249702947.sock)
2016-06-02 00:04:59.652444 I | rafthttp: started streaming with peer 94e5d7e43cbbf7c7 (writer)
2016-06-02 00:04:59.655810 I | rafthttp: started streaming with peer 94e5d7e43cbbf7c7 (writer)
2016-06-02 00:04:59.655940 I | rafthttp: started streaming with peer 94e5d7e43cbbf7c7 (stream MsgApp v2 reader)
2016-06-02 00:04:59.657534 I | rafthttp: peer 8996f73a60a6d61f became active
2016-06-02 00:04:59.657607 I | rafthttp: established a TCP streaming connection with peer 8996f73a60a6d61f (stream Message writer)
2016-06-02 00:04:59.657683 I | rafthttp: peer bf1facf945c123f0 became active
2016-06-02 00:04:59.658031 I | rafthttp: peer 94e5d7e43cbbf7c7 became active
2016-06-02 00:04:59.658095 I | rafthttp: established a TCP streaming connection with peer 94e5d7e43cbbf7c7 (stream Message writer)
2016-06-02 00:04:59.658574 I | rafthttp: established a TCP streaming connection with peer bf1facf945c123f0 (stream Message writer)
2016-06-02 00:04:59.660271 I | rafthttp: established a TCP streaming connection with peer 94e5d7e43cbbf7c7 (stream MsgApp v2 writer)
2016-06-02 00:04:59.660617 I | raft: bf1facf945c123f0 [logterm: 1, index: 3, vote: 0] ignored vote from 8996f73a60a6d61f [logterm: 1, index: 3] at term 1: lease is not expired (remaining ticks: 1)
2016-06-02 00:04:59.660757 I | raft: bf1facf945c123f0 is starting a new election at term 1
2016-06-02 00:04:59.660877 I | raft: bf1facf945c123f0 became candidate at term 2
2016-06-02 00:04:59.660935 I | raft: bf1facf945c123f0 received vote from bf1facf945c123f0 at term 2
2016-06-02 00:04:59.660994 I | raft: bf1facf945c123f0 [logterm: 1, index: 3] sent vote request to 8996f73a60a6d61f at term 2
2016-06-02 00:04:59.661055 I | raft: bf1facf945c123f0 [logterm: 1, index: 3] sent vote request to 94e5d7e43cbbf7c7 at term 2
2016-06-02 00:04:59.661382 I | raft: 94e5d7e43cbbf7c7 [logterm: 1, index: 3, vote: 94e5d7e43cbbf7c7] rejected vote from 8996f73a60a6d61f [logterm: 1, index: 3] at term 2
2016-06-02 00:04:59.662097 I | rafthttp: started streaming with peer 94e5d7e43cbbf7c7 (stream Message reader)
2016-06-02 00:04:59.662385 I | rafthttp: established a TCP streaming connection with peer 8996f73a60a6d61f (stream MsgApp v2 writer)
2016-06-02 00:04:59.662426 I | rafthttp: established a TCP streaming connection with peer 8996f73a60a6d61f (stream Message writer)
2016-06-02 00:04:59.662903 I | rafthttp: peer bf1facf945c123f0 became active
2016-06-02 00:04:59.662982 I | rafthttp: established a TCP streaming connection with peer bf1facf945c123f0 (stream MsgApp v2 reader)
2016-06-02 00:04:59.664286 I | rafthttp: established a TCP streaming connection with peer 8996f73a60a6d61f (stream MsgApp v2 writer)
2016-06-02 00:04:59.664546 I | membership: added member 8996f73a60a6d61f [http://127.0.0.1:21113] to cluster 2b55eb8bf5655e7d
2016-06-02 00:04:59.664834 I | raft: bf1facf945c123f0 [logterm: 1, index: 3, vote: bf1facf945c123f0] rejected vote from 94e5d7e43cbbf7c7 [logterm: 1, index: 3] at term 2
2016-06-02 00:04:59.665124 I | membership: added member 94e5d7e43cbbf7c7 [http://127.0.0.1:21109] to cluster 2b55eb8bf5655e7d
2016-06-02 00:04:59.665471 I | integration: launched node1077664708996568847 (unix://localhost:node1077664708996568847.sock)
2016-06-02 00:04:59.666175 I | rafthttp: established a TCP streaming connection with peer bf1facf945c123f0 (stream Message reader)
2016-06-02 00:04:59.666607 I | rafthttp: established a TCP streaming connection with peer 8996f73a60a6d61f (stream Message reader)
2016-06-02 00:04:59.666951 I | membership: added member bf1facf945c123f0 [http://127.0.0.1:21111] to cluster 2b55eb8bf5655e7d
2016-06-02 00:04:59.667350 I | rafthttp: established a TCP streaming connection with peer bf1facf945c123f0 (stream MsgApp v2 writer)
2016-06-02 00:04:59.667529 I | raft: 94e5d7e43cbbf7c7 [logterm: 1, index: 3, vote: 94e5d7e43cbbf7c7] rejected vote from bf1facf945c123f0 [logterm: 1, index: 3] at term 2
2016-06-02 00:04:59.667693 I | rafthttp: established a TCP streaming connection with peer 94e5d7e43cbbf7c7 (stream Message reader)
2016-06-02 00:04:59.667745 I | rafthttp: established a TCP streaming connection with peer 94e5d7e43cbbf7c7 (stream MsgApp v2 reader)
2016-06-02 00:04:59.667889 I | raft: 8996f73a60a6d61f received vote rejection from 94e5d7e43cbbf7c7 at term 2
2016-06-02 00:04:59.667953 I | raft: 8996f73a60a6d61f [quorum:2] has received 1 votes and 1 vote rejections
2016-06-02 00:04:59.668659 I | rafthttp: established a TCP streaming connection with peer bf1facf945c123f0 (stream Message reader)
2016-06-02 00:04:59.669043 I | raft: 8996f73a60a6d61f [logterm: 1, index: 3, vote: 8996f73a60a6d61f] rejected vote from bf1facf945c123f0 [logterm: 1, index: 3] at term 2
2016-06-02 00:04:59.669365 I | rafthttp: established a TCP streaming connection with peer bf1facf945c123f0 (stream MsgApp v2 reader)
2016-06-02 00:04:59.670197 I | rafthttp: established a TCP streaming connection with peer bf1facf945c123f0 (stream Message writer)
2016-06-02 00:04:59.670397 I | rafthttp: established a TCP streaming connection with peer 94e5d7e43cbbf7c7 (stream Message reader)
2016-06-02 00:04:59.670589 I | raft: bf1facf945c123f0 received vote rejection from 8996f73a60a6d61f at term 2
2016-06-02 00:04:59.670800 I | raft: bf1facf945c123f0 [quorum:2] has received 1 votes and 1 vote rejections
2016-06-02 00:04:59.671164 I | raft: 8996f73a60a6d61f [logterm: 1, index: 3, vote: 8996f73a60a6d61f] rejected vote from 94e5d7e43cbbf7c7 [logterm: 1, index: 3] at term 2
2016-06-02 00:04:59.671228 I | raft: 94e5d7e43cbbf7c7 received vote rejection from bf1facf945c123f0 at term 2
2016-06-02 00:04:59.671278 I | raft: 94e5d7e43cbbf7c7 [quorum:2] has received 1 votes and 1 vote rejections
2016-06-02 00:04:59.671371 I | rafthttp: established a TCP streaming connection with peer 8996f73a60a6d61f (stream MsgApp v2 reader)
2016-06-02 00:04:59.672349 I | rafthttp: established a TCP streaming connection with peer bf1facf945c123f0 (stream MsgApp v2 writer)
2016-06-02 00:04:59.672475 I | raft: bf1facf945c123f0 received vote rejection from 94e5d7e43cbbf7c7 at term 2
2016-06-02 00:04:59.672528 I | raft: bf1facf945c123f0 [quorum:2] has received 1 votes and 2 vote rejections
2016-06-02 00:04:59.672932 I | raft: bf1facf945c123f0 became follower at term 2
2016-06-02 00:04:59.673172 I | raft: 94e5d7e43cbbf7c7 received vote rejection from 8996f73a60a6d61f at term 2
2016-06-02 00:04:59.673220 I | raft: 94e5d7e43cbbf7c7 [quorum:2] has received 1 votes and 2 vote rejections
2016-06-02 00:04:59.673279 I | raft: 94e5d7e43cbbf7c7 became follower at term 2
2016-06-02 00:04:59.673397 I | rafthttp: established a TCP streaming connection with peer 94e5d7e43cbbf7c7 (stream MsgApp v2 reader)
2016-06-02 00:04:59.798503 I | raft: bf1facf945c123f0 is starting a new election at term 2
2016-06-02 00:04:59.798591 I | raft: bf1facf945c123f0 became candidate at term 3
2016-06-02 00:04:59.798629 I | raft: bf1facf945c123f0 received vote from bf1facf945c123f0 at term 3
2016-06-02 00:04:59.798673 I | raft: bf1facf945c123f0 [logterm: 1, index: 3] sent vote request to 8996f73a60a6d61f at term 3
2016-06-02 00:04:59.798714 I | raft: bf1facf945c123f0 [logterm: 1, index: 3] sent vote request to 94e5d7e43cbbf7c7 at term 3
2016-06-02 00:04:59.799110 I | raft: 94e5d7e43cbbf7c7 [term: 2] received a MsgVote message with higher term from bf1facf945c123f0 [term: 3]
2016-06-02 00:04:59.799144 I | raft: 8996f73a60a6d61f [term: 2] received a MsgVote message with higher term from bf1facf945c123f0 [term: 3]
2016-06-02 00:04:59.799182 I | raft: 94e5d7e43cbbf7c7 became follower at term 3
2016-06-02 00:04:59.799211 I | raft: 8996f73a60a6d61f became follower at term 3
2016-06-02 00:04:59.799254 I | raft: 94e5d7e43cbbf7c7 [logterm: 1, index: 3, vote: 0] voted for bf1facf945c123f0 [logterm: 1, index: 3] at term 3
2016-06-02 00:04:59.799417 I | raft: 8996f73a60a6d61f [logterm: 1, index: 3, vote: 0] voted for bf1facf945c123f0 [logterm: 1, index: 3] at term 3
2016-06-02 00:04:59.799640 I | raft: bf1facf945c123f0 received vote from 94e5d7e43cbbf7c7 at term 3
2016-06-02 00:04:59.799677 I | raft: bf1facf945c123f0 [quorum:2] has received 2 votes and 0 vote rejections
2016-06-02 00:04:59.799738 I | raft: bf1facf945c123f0 became leader at term 3
2016-06-02 00:04:59.799800 I | raft: raft.node: bf1facf945c123f0 elected leader bf1facf945c123f0 at term 3
2016-06-02 00:04:59.800231 I | raft: raft.node: 8996f73a60a6d61f elected leader bf1facf945c123f0 at term 3
2016-06-02 00:04:59.800377 I | raft: raft.node: 94e5d7e43cbbf7c7 elected leader bf1facf945c123f0 at term 3
2016-06-02 00:04:59.803121 I | etcdserver: published {Name:node1077664708996568847 ClientURLs:[http://127.0.0.1:21112]} to cluster 2b55eb8bf5655e7d
2016-06-02 00:04:59.804152 I | etcdserver: published {Name:node6962765416249702947 ClientURLs:[http://127.0.0.1:21114]} to cluster 2b55eb8bf5655e7d
2016-06-02 00:04:59.804423 I | etcdserver: published {Name:node7564707074000347284 ClientURLs:[http://127.0.0.1:21110]} to cluster 2b55eb8bf5655e7d
2016-06-02 00:04:59.805364 I | etcdserver: setting up the initial cluster version to 3.0
2016-06-02 00:04:59.807236 N | membership: set the initial cluster version to 3.0
2016-06-02 00:04:59.807869 N | membership: set the initial cluster version to 3.0
2016-06-02 00:04:59.808144 N | membership: set the initial cluster version to 3.0
2016-06-02 00:04:59.811101 I | integration: stopping node7564707074000347284 (unix://localhost:node7564707074000347284.sock)
2016-06-02 00:04:59.811565 I | rafthttp: stopping peer 8996f73a60a6d61f...
2016-06-02 00:04:59.812382 I | rafthttp: closed the TCP streaming connection with peer 8996f73a60a6d61f (stream MsgApp v2 writer)
2016-06-02 00:04:59.812478 I | rafthttp: stopped streaming with peer 8996f73a60a6d61f (writer)
2016-06-02 00:04:59.812540 W | rafthttp: lost the TCP streaming connection with peer 94e5d7e43cbbf7c7 (stream MsgApp v2 reader)
2016-06-02 00:04:59.812802 W | rafthttp: lost the TCP streaming connection with peer 94e5d7e43cbbf7c7 (stream Message reader)
2016-06-02 00:04:59.813405 I | rafthttp: closed the TCP streaming connection with peer 8996f73a60a6d61f (stream Message writer)
2016-06-02 00:04:59.813455 I | rafthttp: stopped streaming with peer 8996f73a60a6d61f (writer)
2016-06-02 00:04:59.813538 I | rafthttp: stopped HTTP pipelining with peer 8996f73a60a6d61f
2016-06-02 00:04:59.813675 W | rafthttp: lost the TCP streaming connection with peer 8996f73a60a6d61f (stream MsgApp v2 reader)
2016-06-02 00:04:59.813733 E | rafthttp: failed to read 8996f73a60a6d61f on stream MsgApp v2 (net/http: request canceled)
2016-06-02 00:04:59.813762 I | rafthttp: peer 8996f73a60a6d61f became inactive
2016-06-02 00:04:59.813803 I | rafthttp: stopped streaming with peer 8996f73a60a6d61f (stream MsgApp v2 reader)
2016-06-02 00:04:59.813926 W | rafthttp: lost the TCP streaming connection with peer 8996f73a60a6d61f (stream Message reader)
2016-06-02 00:04:59.814002 I | rafthttp: stopped streaming with peer 8996f73a60a6d61f (stream Message reader)
2016-06-02 00:04:59.814050 I | rafthttp: stopped peer 8996f73a60a6d61f
2016-06-02 00:04:59.814088 I | rafthttp: stopping peer bf1facf945c123f0...
2016-06-02 00:04:59.814612 W | rafthttp: lost the TCP streaming connection with peer 94e5d7e43cbbf7c7 (stream MsgApp v2 reader)
2016-06-02 00:04:59.814995 I | rafthttp: closed the TCP streaming connection with peer bf1facf945c123f0 (stream MsgApp v2 writer)
2016-06-02 00:04:59.815053 I | rafthttp: stopped streaming with peer bf1facf945c123f0 (writer)
2016-06-02 00:04:59.815498 W | rafthttp: lost the TCP streaming connection with peer 94e5d7e43cbbf7c7 (stream Message reader)
2016-06-02 00:04:59.815923 I | rafthttp: closed the TCP streaming connection with peer bf1facf945c123f0 (stream Message writer)
2016-06-02 00:04:59.816144 I | rafthttp: stopped streaming with peer bf1facf945c123f0 (writer)
2016-06-02 00:04:59.816304 I | rafthttp: stopped HTTP pipelining with peer bf1facf945c123f0
2016-06-02 00:04:59.816523 W | rafthttp: lost the TCP streaming connection with peer bf1facf945c123f0 (stream MsgApp v2 reader)
2016-06-02 00:04:59.817418 E | rafthttp: failed to read bf1facf945c123f0 on stream MsgApp v2 (net/http: request canceled)
2016-06-02 00:04:59.817483 I | rafthttp: peer bf1facf945c123f0 became inactive
2016-06-02 00:04:59.817564 I | rafthttp: stopped streaming with peer bf1facf945c123f0 (stream MsgApp v2 reader)
2016-06-02 00:04:59.817727 W | rafthttp: lost the TCP streaming connection with peer bf1facf945c123f0 (stream Message reader)
2016-06-02 00:04:59.817820 I | rafthttp: stopped streaming with peer bf1facf945c123f0 (stream Message reader)
2016-06-02 00:04:59.817876 I | rafthttp: stopped peer bf1facf945c123f0
2016-06-02 00:04:59.826118 I | integration: stopped node7564707074000347284 (unix://localhost:node7564707074000347284.sock)
[DEBUG] 2016-06-02 00:04:59.826165268 +0000 UTC kv.Txn stop done!
[DEBUG] 2016-06-02 00:04:59.826239284 +0000 UTC kv.Txn starting!
2016-06-02 00:04:59.848624 E | rafthttp: failed to write 94e5d7e43cbbf7c7 on stream Message (write tcp 127.0.0.1:21111->127.0.0.1:35833: write: broken pipe)
2016-06-02 00:04:59.848679 I | rafthttp: peer 94e5d7e43cbbf7c7 became inactive
2016-06-02 00:04:59.849499 W | rafthttp: lost the TCP streaming connection with peer 94e5d7e43cbbf7c7 (stream Message writer)
2016-06-02 00:04:59.913138 E | rafthttp: failed to dial 94e5d7e43cbbf7c7 on stream MsgApp v2 (dial tcp 127.0.0.1:21109: getsockopt: connection refused)
2016-06-02 00:04:59.913198 I | rafthttp: peer 94e5d7e43cbbf7c7 became inactive
2016-06-02 00:05:01.318176 W | rafthttp: lost the TCP streaming connection with peer 94e5d7e43cbbf7c7 (stream MsgApp v2 writer)
2016-06-02 00:05:01.331652 W | etcdserver: failed to send out heartbeat on time (exceeded the 10ms timeout for 3.050851ms)
2016-06-02 00:05:01.331745 W | etcdserver: server is likely overloaded
2016-06-02 00:05:01.331794 W | etcdserver: failed to send out heartbeat on time (exceeded the 10ms timeout for 3.208878ms)
2016-06-02 00:05:01.331825 W | etcdserver: server is likely overloaded
2016-06-02 00:05:02.254015 W | etcdserver: failed to send out heartbeat on time (exceeded the 10ms timeout for 5.275378ms)
2016-06-02 00:05:02.254089 W | etcdserver: server is likely overloaded
2016-06-02 00:05:02.257564 W | etcdserver: failed to send out heartbeat on time (exceeded the 10ms timeout for 8.836674ms)
2016-06-02 00:05:02.261354 W | etcdserver: server is likely overloaded
2016-06-02 00:05:03.807011 W | etcdserver: failed to reach the peerURL(http://127.0.0.1:21109) of member 94e5d7e43cbbf7c7 (Get http://127.0.0.1:21109/version: dial tcp 127.0.0.1:21109: getsockopt: connection refused)
2016-06-02 00:05:03.810441 W | etcdserver: cannot get the version of member 94e5d7e43cbbf7c7 (Get http://127.0.0.1:21109/version: dial tcp 127.0.0.1:21109: getsockopt: connection refused)
2016-06-02 00:05:06.276969 W | rafthttp: lost the TCP streaming connection with peer 94e5d7e43cbbf7c7 (stream MsgApp v2 writer)
2016-06-02 00:05:06.277258 W | rafthttp: lost the TCP streaming connection with peer 94e5d7e43cbbf7c7 (stream Message writer)
[DEBUG] 2016-06-02 00:05:07.379072576 +0000 UTC connWait c.lastConnErr grpc: timed out trying to connect
2016-06-02 00:05:07.811783 W | etcdserver: failed to reach the peerURL(http://127.0.0.1:21109) of member 94e5d7e43cbbf7c7 (Get http://127.0.0.1:21109/version: dial tcp 127.0.0.1:21109: getsockopt: connection refused)
2016-06-02 00:05:07.811837 W | etcdserver: cannot get the version of member 94e5d7e43cbbf7c7 (Get http://127.0.0.1:21109/version: dial tcp 127.0.0.1:21109: getsockopt: connection refused)
[DEBUG] 2016-06-02 00:05:10.826283022 +0000 UTC timed out waiting for txn to fail
2016-06-02 00:05:11.813448 W | etcdserver: failed to reach the peerURL(http://127.0.0.1:21109) of member 94e5d7e43cbbf7c7 (Get http://127.0.0.1:21109/version: dial tcp 127.0.0.1:21109: getsockopt: connection refused)
2016-06-02 00:05:11.813525 W | etcdserver: cannot get the version of member 94e5d7e43cbbf7c7 (Get http://127.0.0.1:21109/version: dial tcp 127.0.0.1:21109: getsockopt: connection refused)
[DEBUG] 2016-06-02 00:05:11.90833063 +0000 UTC connWait c.lastConnErr grpc: timed out trying to connect
[DEBUG] 2016-06-02 00:05:11.908408404 +0000 UTC kv.Txn error: rpc error: code = 2 desc = grpc: the client connection is closing
2016-06-02 00:05:11.909870 I | integration: terminating node7564707074000347284 (unix://localhost:node7564707074000347284.sock)
2016-06-02 00:05:11.916167 I | integration: terminated node7564707074000347284 (unix://localhost:node7564707074000347284.sock)
2016-06-02 00:05:11.916258 I | integration: terminating node1077664708996568847 (unix://localhost:node1077664708996568847.sock)
2016-06-02 00:05:11.916490 I | rafthttp: stopping peer 8996f73a60a6d61f...
2016-06-02 00:05:11.916918 W | rafthttp: lost the TCP streaming connection with peer bf1facf945c123f0 (stream MsgApp v2 reader)
2016-06-02 00:05:11.917368 I | rafthttp: closed the TCP streaming connection with peer 8996f73a60a6d61f (stream MsgApp v2 writer)
2016-06-02 00:05:11.917426 I | rafthttp: stopped streaming with peer 8996f73a60a6d61f (writer)
2016-06-02 00:05:11.918294 I | rafthttp: closed the TCP streaming connection with peer 8996f73a60a6d61f (stream Message writer)
2016-06-02 00:05:11.918453 I | rafthttp: stopped streaming with peer 8996f73a60a6d61f (writer)
2016-06-02 00:05:11.918552 I | rafthttp: stopped HTTP pipelining with peer 8996f73a60a6d61f
2016-06-02 00:05:11.918748 W | rafthttp: lost the TCP streaming connection with peer 8996f73a60a6d61f (stream MsgApp v2 reader)
2016-06-02 00:05:11.918813 I | rafthttp: stopped streaming with peer 8996f73a60a6d61f (stream MsgApp v2 reader)
2016-06-02 00:05:11.918987 W | rafthttp: lost the TCP streaming connection with peer 8996f73a60a6d61f (stream Message reader)
2016-06-02 00:05:11.919041 I | rafthttp: stopped streaming with peer 8996f73a60a6d61f (stream Message reader)
2016-06-02 00:05:11.919083 I | rafthttp: stopped peer 8996f73a60a6d61f
2016-06-02 00:05:11.919116 I | rafthttp: stopping peer 94e5d7e43cbbf7c7...
2016-06-02 00:05:11.919197 I | rafthttp: stopped streaming with peer 94e5d7e43cbbf7c7 (writer)
2016-06-02 00:05:11.919337 I | rafthttp: stopped streaming with peer 94e5d7e43cbbf7c7 (writer)
2016-06-02 00:05:11.919603 I | rafthttp: stopped HTTP pipelining with peer 94e5d7e43cbbf7c7
2016-06-02 00:05:11.919742 I | rafthttp: stopped streaming with peer 94e5d7e43cbbf7c7 (stream MsgApp v2 reader)
2016-06-02 00:05:11.919900 I | rafthttp: stopped streaming with peer 94e5d7e43cbbf7c7 (stream Message reader)
2016-06-02 00:05:11.920023 I | rafthttp: stopped peer 94e5d7e43cbbf7c7
2016-06-02 00:05:11.923003 W | rafthttp: lost the TCP streaming connection with peer bf1facf945c123f0 (stream Message reader)
2016-06-02 00:05:11.935943 I | integration: terminated node1077664708996568847 (unix://localhost:node1077664708996568847.sock)
2016-06-02 00:05:11.936023 I | integration: terminating node6962765416249702947 (unix://localhost:node6962765416249702947.sock)
2016-06-02 00:05:11.936293 I | rafthttp: stopping peer 94e5d7e43cbbf7c7...
2016-06-02 00:05:11.936426 I | rafthttp: stopped streaming with peer 94e5d7e43cbbf7c7 (writer)
2016-06-02 00:05:11.936522 I | rafthttp: stopped streaming with peer 94e5d7e43cbbf7c7 (writer)
2016-06-02 00:05:11.936690 I | rafthttp: stopped HTTP pipelining with peer 94e5d7e43cbbf7c7
2016-06-02 00:05:11.936761 I | rafthttp: stopped streaming with peer 94e5d7e43cbbf7c7 (stream MsgApp v2 reader)
2016-06-02 00:05:11.936817 I | rafthttp: stopped streaming with peer 94e5d7e43cbbf7c7 (stream Message reader)
2016-06-02 00:05:11.936854 I | rafthttp: stopped peer 94e5d7e43cbbf7c7
2016-06-02 00:05:11.936881 I | rafthttp: stopping peer bf1facf945c123f0...
2016-06-02 00:05:11.937708 I | rafthttp: closed the TCP streaming connection with peer bf1facf945c123f0 (stream MsgApp v2 writer)
2016-06-02 00:05:11.937747 I | rafthttp: stopped streaming with peer bf1facf945c123f0 (writer)
2016-06-02 00:05:11.938609 I | rafthttp: closed the TCP streaming connection with peer bf1facf945c123f0 (stream Message writer)
2016-06-02 00:05:11.938677 I | rafthttp: stopped streaming with peer bf1facf945c123f0 (writer)
2016-06-02 00:05:11.938841 I | rafthttp: stopped HTTP pipelining with peer bf1facf945c123f0
2016-06-02 00:05:11.938927 I | rafthttp: stopped streaming with peer bf1facf945c123f0 (stream MsgApp v2 reader)
2016-06-02 00:05:11.939010 I | rafthttp: stopped streaming with peer bf1facf945c123f0 (stream Message reader)
2016-06-02 00:05:11.939076 I | rafthttp: stopped peer bf1facf945c123f0
2016-06-02 00:05:11.952468 I | integration: terminated node6962765416249702947 (unix://localhost:node6962765416249702947.sock)
--- FAIL: TestTxnWriteFail (12.41s)
	txn_test.go:83: timed out waiting for txn to fail
```

Thanks for help!

xiang90 commented:
Where is your patched version of client? I am not clear where you printed out the debug.

gyuho commented:
@xiang90 Here's the link to my patch https://github.com/coreos/etcd/pull/5523/files.

xiang90 commented:
OK. I reproduced this myself. I am pretty sure that we should not make the timeout longer. Txn is supposed to fail fast. This is what this test is for, we should not make the timeout longer for no good reason. See https://github.com/coreos/etcd/issues/5416#issuecomment-223362955

gyuho commented:
got it. thanks.
